### PR TITLE
Allow timezone suffix in the date-time regexp

### PR DESCRIPTION
**Pull request by honzajavorek**

### CODE REVIEW

**Comment by kylef** [comment]
Would be great to add some tests, there is some for `date` (https://github.com/apiaryio/Amanda/blob/master/tests/json/attributes/format/date.js), should be trivial to add same for date-time including a test for the proposed fix (and to make sure there is no regression for other formats that are meant to be supported).

There's a good testset used in other JSON Schema libraries too, there's some examples you could use: https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/9567c588bc9d1705b4840ee9d9f3f268f40c1e68/tests/draft4/optional/format.json#L3 although it isn't that complete, copying a bunch of the examples from RFC3339 would probably be a good start.

---

I was curious, is this library used in Dredd and that is where the bug is surfacing to the user?

**Comment by honzajavorek** [comment]
This library is used to validate draft3 schemas in Dredd. I'd love to deprecate it one day.

**Comment by honzajavorek** [comment]
And I tested on https://regex101.com/ :trollface: But yeah, you're right, tests are missing.

**Comment by michalholasek** [comment]
@honzajavorek Sooo, will you add the tests or not? :>

**Comment by honzajavorek** [comment]
Yup, I will

**Comment by honzajavorek** [comment]
@michalholasek done

File: tests/json/attributes/format/date-time.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+if (typeof module !== 'undefined' && module.exports) {
+  var expect = require('expect.js');
+  var amanda = require('../../../../releases/latest/amanda.js');
+}
+
+/**
+ * Format
+ * --------------------
+ */
+suite('JSON/Attribute/format#date-time', function() {
```
Comment (michalholasek) [inline tests/json/attributes/format/date-time.js:10]: Inconsistent `function()` and `function ()` - please choose one and stick with it (I think `function()` is preferred in other test files).

**Review by michalholasek — COMMENTED** [review]
One small [nitpick](https://github.com/apiaryio/Amanda/pull/9/files#r194735917), otherwise looks good to me 👍

File: tests/json/attributes/format/date-time.js (hunk 1 above)
Comment (honzajavorek) [reply tests/json/attributes/format/date-time.js:10]: Oh, this was plain copy-paste. I think VS Code had to do some autocorrections.

**Pull request merged**